modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -2433,7 +2434,7 @@ private T doPut(T row, boolean needOld) throws IgniteCheckedException {
                 }
             }
         }
-        catch (IgniteCheckedException e) {
+        catch (IgniteCheckedException | IgniteSQLException e) {
```

Comment:
Can you please remove this new exception from here? I believe that it's unused and left from previous fix attempt. Thank you!